ui/src/components/ShipConnection.tsx

```diff
@@ -0,0 +1,100 @@
+import cn from 'classnames';
+import React from 'react';
+import {
+  ConnectionCompleteStatus,
+  ConnectionPendingStatus,
+  ConnectionStatus,
+} from '../state/vitals';
+import Bullet16Icon from './icons/Bullet16Icon';
+
+interface ShipConnectionProps {
+  ship: string;
+  status?: ConnectionStatus;
+  showBullet?: boolean;
+  showText?: boolean;
+  className?: string;
+}
+
+function getCompletedText(status: ConnectionCompleteStatus, ship: string) {
```

Comment:
We may want to move these functions into utils

Reply:
Handled, chef!